rllib/agents/ddpg/ddpg_tf_policy.py

```diff
@@ -316,10 +310,19 @@ def gradients_fn(policy, optimizer, loss):
             var_list=policy.model.q_variables(),
             clip_val=policy.config["grad_norm_clipping"])
     else:
-        actor_grads_and_vars = policy._actor_optimizer.compute_gradients(
-            policy.actor_loss, var_list=policy.model.policy_variables())
-        critic_grads_and_vars = policy._critic_optimizer.compute_gradients(
-            policy.critic_loss, var_list=policy.model.q_variables())
+        if policy.config["framework"] == "tfe":
+            tape = optimizer.tape
```

Comment:
Why not use a simple optimizer.get_gradients call? Most people are not too familiar with optimizer.tape.

Reply:
+1, let's try that. ...

Reply:
See below:
```
It's more problematic, I think:
We have three optimizers for the graph-mode case, but only one (faked by the caller of gradients_fn and carrying the tape object) for the eager case.
You could create a dict/tuple (each optimizer holding grad-fn plus vars), but I think that's even uglier :( .
```